client/app/(auth)/createProfile.tsx

```diff
@@ -1,19 +1,430 @@
+  }, [currentUser, reset]);
+
+  const handleCreateProfile = async (model: UserEditModel) => {
+    setIsLoading(true);
+    try {
+      if (currentUser) {
+        await UserController.editUser(currentUser.uid, model);
+        const userData = await UserController.getUser(currentUser.uid);
+        dispatch(setUser(userData));
+        router.push("/(auth)/createTeam");
+      } else {
+        router.push("/(auth)");
+      }
+    } catch (error) {
+      console.log(error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
   return (
-    <View className="flex-1 items-center justify-center p-5">
-      <Text className="pbk-h6 text-gray-950">This is temporary</Text>
-      <Text className="pbk-b1 text-gray-950">{user.userId}</Text>
-      <Text className="pbk-b1 text-gray-950">{user.email}</Text>
-      <Text className="pbk-b1 text-gray-950">{user.username}</Text>
-
-      <Link className="mt-4 py-4" href="/">
-        <Text className="pbk-b2 text-purple-500">Go to home screen!</Text>
-      </Link>
+    <View className="flex-1 bg-gray-950">
+      <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+        <View className="flex-1">
+          <KeyboardAvoidingView behavior="padding" className="flex-1">
+            <View className="flex-1 flex-col px-6 py-4">
+              <View className="my-8 size-8 self-end" />
+
+              <Text className="pbk-h5 mb-8 text-base-white">
+                Tell us who you are
+              </Text>
+
+              <Text className="pbk-b2 mb-1.5 text-base-white">Username</Text>
+              <Controller
+                control={control}
+                name="username"
+                render={({ field: { onChange, value } }) => (
+                  <View
+                    className={`mb-2 min-h-14 w-full flex-row items-center rounded-xl border ${errors.username ? "border-red-600" : "border-gray-920"} px-2 py-2`}
+                  >
+                    <TextInput
+                      autoCapitalize="none"
+                      className="flex-1 text-base-white placeholder:pbk-b1"
+                      onChangeText={(text) => {
+                        onChange(text);
+                      }}
+                      placeholder="Enter your username"
+                      placeholderTextColor="gray"
+                      value={value}
+                    />
+                    {errors.username && (
+                      <WarningCircle color="#dc2626" size={20} weight="bold" />
+                    )}
+                  </View>
+                )}
+              />
+              {errors.username && (
+                <Text className="pbk-b3 mb-4 text-red-600">
+                  {errors.username.message}
+                </Text>
+              )}
+              <View className="mb-5">
+                <Text className="pbk-b3 text-gray-600">
+                  This is your display name. You can change it later.
+                </Text>
+              </View>
```

Comment:
I think we should include the error message into this View you have set up but also create a conditional render for the message so we don't display too many notes. When there is an error that occurs, it should prioritize letting the user know about that error and then go back to the default text. This also fixes some of the spacing issues I think as I changed the margin Bottom of this View to mirror the spacing between the other TextInputs and labels.

```suggestion            
              <View className="mb-4">
                {errors.username ? (
                  <Text className="pbk-b3 text-red-600">
                    {errors.username.message}
                  </Text>
                ) : (
                  <Text className="pbk-b3 text-gray-600">
                    This is your display name. You can change it later.
                  </Text>
                )}
              </View>
```

Reply:
Yeah nice, this works well! Changed now